src/python/grpcio_tests/tests_aio/unit/server_test.py

```diff
@@ -98,89 +174,167 @@ def service(self, handler_details):
-        )
+        addr, self._server, self._generic_handler = await _start_test_server()
+        self._channel = aio.insecure_channel(addr)
 
     async def tearDown(self):
+        await self._channel.close()
         await self._server.stop(None)
 
     async def test_unary_unary(self):
-        async with aio.insecure_channel(self._server_target) as channel:
-            unary_unary_call = channel.unary_unary(_SIMPLE_UNARY_UNARY)
-            response = await unary_unary_call(_REQUEST)
-            self.assertEqual(response, _RESPONSE)
+        unary_unary_call = self._channel.unary_unary(_SIMPLE_UNARY_UNARY)
+        response = await unary_unary_call(_REQUEST)
+        self.assertEqual(response, _RESPONSE)
 
     async def test_unary_stream_async_generator(self):
-        async with aio.insecure_channel(self._server_target) as channel:
-            unary_stream_call = channel.unary_stream(_UNARY_STREAM_ASYNC_GEN)
-            call = unary_stream_call(_REQUEST)
+        unary_stream_call = self._channel.unary_stream(_UNARY_STREAM_ASYNC_GEN)
+        call = unary_stream_call(_REQUEST)
 
-            # Expecting the request message to reach server before retriving
-            # any responses.
-            await asyncio.wait_for(self._generic_handler.wait_for_call(),
-                                   test_constants.SHORT_TIMEOUT)
+        # Expecting the request message to reach server before retriving
+        # any responses.
+        await asyncio.wait_for(self._generic_handler.wait_for_call(),
+                               test_constants.SHORT_TIMEOUT)
 
-            response_cnt = 0
-            async for response in call:
-                response_cnt += 1
-                self.assertEqual(_RESPONSE, response)
+        response_cnt = 0
+        async for response in call:
+            response_cnt += 1
+            self.assertEqual(_RESPONSE, response)
 
-            self.assertEqual(_NUM_STREAM_RESPONSES, response_cnt)
-            self.assertEqual(await call.code(), grpc.StatusCode.OK)
+        self.assertEqual(_NUM_STREAM_RESPONSES, response_cnt)
+        self.assertEqual(await call.code(), grpc.StatusCode.OK)
 
     async def test_unary_stream_reader_writer(self):
-        async with aio.insecure_channel(self._server_target) as channel:
-            unary_stream_call = channel.unary_stream(
-                _UNARY_STREAM_READER_WRITER)
-            call = unary_stream_call(_REQUEST)
+        unary_stream_call = self._channel.unary_stream(
+            _UNARY_STREAM_READER_WRITER)
+        call = unary_stream_call(_REQUEST)
 
-            # Expecting the request message to reach server before retriving
-            # any responses.
-            await asyncio.wait_for(self._generic_handler.wait_for_call(),
-                                   test_constants.SHORT_TIMEOUT)
+        # Expecting the request message to reach server before retriving
```

Comment:
Same question as above. I don't see why this is necessary.

Reply:
Your are right. They are not necessary.